src/utils/filters.js

```diff
@@ -0,0 +1,23 @@
+import { enumMsg } from '_/intl'
+
+const compareMap = {
+  name: (item, filter) => !!filter.find(n => item.get('name').startsWith(n)),
+  os: (item, filter) => item.getIn(['os', 'type']) === filter,
+  status: (item, filter) => item.get('status') === filter,
+}
+
+export const mapFilterValues = {
+  name: (value) => value,
+  os: (value) => value,
+  status: (value) => enumMsg('VmStatus', value),
+}
+
+export function filterVms (item, filters) {
+  let res = true
+  for (let name in filters) {
+    if (compareMap[name]) {
+      res &= compareMap[name](item, filters[name])
```

Comment:
Did you intend to use the bitwise AND here or just letting the value be coerced between boolean and numerics?

Reply:
I'm not sure what you mean, but values returned by `compareMap` functions are boolean, so there is no coercing.

Reply:
`&=` is a bitwise operation not a boolean operation.  

The values of `res` on the way out will be true if no filters are applied, or numeric 0 or 1 if filters are applied.  It'll still be truthy correct, but it won't be a boolean value.